golangServer/api/search.go

```diff
@@ -0,0 +1,185 @@
+package api
+
+import (
+	"Oracle.com/golangServer/Oracle"
+	"Oracle.com/golangServer/config"
+	"context"
+	"fmt"
+	"github.com/RangerMauve/ipld-prolly-indexer/indexer"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/node/basicnode"
+	"log"
+)
+
+func SearchEventListener() {
+	// Get channels for logs
+	Logs := make(chan *Oracle.OracleSearch)
+	// Subscribe to each event
+	opts := &bind.WatchOpts{Context: context.Background(), Start: nil}
+	eventSub, err := config.OracleContract.WatchSearch(opts, Logs)
+	if err != nil {
+		log.Fatal("Failed to subscribe to Search events:", err)
+	}
+	// start Listening...
+	log.Println("SearchEvent Listening ...")
+	for {
+		select {
+		case err := <-eventSub.Err():
+			log.Println("[Error in Event SEARCH]:", err)
+		case event := <-Logs:
+			log.Println("Received search event ", event.ReqID)
+			search(event)
+		}
+	}
+}
+
+// Search data from memory db
+func search(event *Oracle.OracleSearch) {
+	var statement bool
+	tps := GenTransactOpts(config.GasLimit)
+	var data []byte
+
+	dbName := event.DbName
+	ctx, dbC := config.Dbs[dbName].Ctx, config.Dbs[dbName].Db
+	method := event.Method
+
+	switch method {
```

Comment:
Is there a way to pass several parameters to search by at once? It feels like it'd be best to allow users to pass the JSON representation of a `indexer.Query` object and to convert that to a real query to perform a search. Especially since `sort`, `limit` and `skip` need to be specified all together for them to work effectively.

Reply:
This works, but when I tried to implement it, I found that I couldn't deserialize `Query` serialized to `json` to `Node `properly. Therefore, I have not implemented this function in this submission.

Reply:
I'll try again tomorrow